src/modules/interface/param.h

```diff
@@ -168,52 +91,69 @@ struct param_s {
 
 #define PARAM_FLOAT (PARAM_4BYTES | PARAM_TYPE_FLOAT | PARAM_SIGNED)
 
+// CRTP
+#define TOC_CH 0
+#define READ_CH 1
+#define WRITE_CH 2
+#define MISC_CH 3
+// CRTP Misc
+#define MISC_SETBYNAME            0
+#define MISC_VALUE_UPDATED        1
+#define MISC_GET_EXTENDED_TYPE    2
+#define MISC_PERSISTENT_STORE     3
+#define MISC_PERSISTENT_GET_STATE 4
+#define MISC_PERSISTENT_CLEAR     5
+#define PERSISTENT_PREFIX_STRING "prm/"
+
 /* Macros */
-#ifndef UNIT_TEST_MODE
+
+#define PARAM_ADD_FULL(TYPE, NAME, ADDRESS, CALLBACK, DEFAULT_GETTER) \
+    { .type = ((TYPE) <= 0xFF) ? (TYPE) : (((TYPE) | PARAM_EXTENDED) & 0xFF), \
+      .extended_type = (((TYPE) & 0xFF00) >> 8), \
+      .name = #NAME, \
+      .address = (void*)(ADDRESS), \
+      .callback = (void *)CALLBACK, \
+      .getter = (void *)DEFAULT_GETTER, },
 
 #define PARAM_ADD(TYPE, NAME, ADDRESS) \
-  { .type = TYPE, .name = #NAME, .address = (void*)(ADDRESS), .callback = 0, },
+    PARAM_ADD_FULL(TYPE, NAME, ADDRESS, 0, 0)
 
 // The callback notification function will run from the param task, it should not block and should run quickly.
 #define PARAM_ADD_WITH_CALLBACK(TYPE, NAME, ADDRESS, CALLBACK) \
-  { .type = TYPE, .name = #NAME, .address = (void*)(ADDRESS), .callback = (void *)CALLBACK, },
+    PARAM_ADD_FULL(TYPE, NAME, ADDRESS, CALLBACK, 0)
 
 #define PARAM_ADD_CORE(TYPE, NAME, ADDRESS) \
   PARAM_ADD(TYPE | PARAM_CORE, NAME, ADDRESS)
 
 #define PARAM_ADD_CORE_WITH_CALLBACK(TYPE, NAME, ADDRESS, CALLBACK) \
   PARAM_ADD_WITH_CALLBACK(TYPE | PARAM_CORE, NAME, ADDRESS, CALLBACK)
 
+#define PARAM_ADD_PERSISTENT(TYPE, NAME, ADDRESS, DEFAULT_GETTER) \
```

Comment:
As previously discussed should we remove this macro and only use PARAM_PERSISTENT flag. A macro might be simpler but there would have to be multiple flavors. If however the persistent param should always be core then a macro would be preferred to enforce this.